flaky test failure in REST

```
=== NAME  TestREST
    service_test.go:99: 
        	Error Trace:	/home/runner/work/rollkit/rollkit/rpc/json/service_test.go:99
        	Error:      	"{\"jsonrpc\":\"2.0\",\"result\":{\"block_id\":{\"hash\":\"CA7F7790577FC09E303816CAC94201914615260D3097D1A04B4BDA0F0377AA45\",\"parts\":{\"total\":0,\"hash\":\"\"}},\"block\":{\"header\":{\"version\":{\"block\":\"11\"},\"chain_id\":\"test\",\"height\":\"1\",\"time\":\"2024-08-14T19:21:25.972545451Z\",\"last_block_id\":{\"hash\":\"\",\"parts\":{\"total\":0,\"hash\":\"\"}},\"last_commit_hash\":\"EA312FCB12D4AEB315427DE61F738D6790DF83E49A83108F1B745947598950B4\",\"data_hash\":\"6E340B9CFFB37A989CA544E6BB780A2C78901D3FB33738768511A30617AFA01D\",\"validators_hash\":\"0554D6F24D8529FF529B4337E519C29516249441C316719E99E1A2D088A4A02B\",\"next_validators_hash\":\"0554D6F24D8529FF529B4337E519C29516249441C316719E99E1A2D088A4A02B\",\"consensus_hash\":\"0000000000000000000000000000000000000000000000000000000000000000\",\"app_hash\":\"\",\"last_results_hash\":\"E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855\",\"evidence_hash\":\"E3B0C44298FC1C149AFBF4C8996FB92427AE41E4649B934CA495991B7852B855\",\"proposer_address\":\"5E77C57BE649671B2DE2B019A910E8FFF1CFC5DF\"},\"data\":{\"txs\":[]},\"evidence\":{\"evidence\":null},\"last_commit\":{\"height\":\"1\",\"round\":0,\"block_id\":{\"hash\":\"CA7F7790577FC09E303816CAC94201914615260D3097D1A04B4BDA0F0377AA45\",\"parts\":{\"total\":0,\"hash\":\"\"}},\"signatures\":[{\"block_id_flag\":2,\"validator_address\":\"5E77C57BE649671B2DE2B019A910E8FFF1CFC5DF\",\"timestamp\":\"2024-08-14T19:21:25.972545451Z\",\"signature\":\"lVbbYWu4WLMo6t0+HaQfobd/Fy6NfjQ+63jlysqxqIunaNHx4jZCrj7rcxguG4a5GUouJefw/ILKpzSlmuvzDw==\"}]}}},\"id\":-1}\n" does not contain "failed to load hash from index"
        	Test:       	TestREST
    service_test.go:103: 
        	Error Trace:	/home/runner/work/rollkit/rollkit/rpc/json/service_test.go:103
        	Error:      	Expected value not to be nil.
        	Test:       	TestREST
=== NAME  TestREST/valid/missing_optional_param
    testing.go:1576: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
--- FAIL: TestREST (0.06s)
```

--- a/rpc/json/service_test.go
+++ b/rpc/json/service_test.go
@@ -64,10 +64,10 @@ func TestREST(t *testing.T) {
 		bodyContains string
 	}{
 
-		{"valid/malformed request", "/block?so{}wrong!", http.StatusOK, int(json2.E_INTERNAL), ``},
+		{"valid/malformed request", "/block?so{}wrong!", http.StatusOK, -1, ``},
 		// to keep test simple, allow returning application error in following case
 		{"invalid/missing required param", "/tx", http.StatusOK, int(json2.E_INVALID_REQ), `missing param 'hash'`},
-		{"valid/missing optional param", "/block", http.StatusOK, int(json2.E_INTERNAL), "failed to load hash from index"},
+		{"valid/missing optional param", "/block", http.StatusOK, -1, ``},
 		{"valid/included block tag param", "/block?height=included", http.StatusOK, int(json2.E_INTERNAL), "failed to load hash from index"},
 		{"valid/no params", "/abci_info", http.StatusOK, -1, `"last_block_height":"345"`},
 		{"valid/int param", "/block?height=321", http.StatusOK, int(json2.E_INTERNAL), "failed to load hash from index"},

--- a/rpc/json/service_test.go
+++ b/rpc/json/service_test.go
@@ -2,9 +2,11 @@ package json
 
 import (
 	"bytes"
+	"context"
 	"encoding/base64"
 	"encoding/hex"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
 	"net/http/httptest"
@@ -64,10 +66,10 @@ func TestREST(t *testing.T) {
 		bodyContains string
 	}{
 
-		{"valid/malformed request", "/block?so{}wrong!", http.StatusOK, -1, ``},
+		{"valid/malformed request", "/block?so{}wrong!", http.StatusOK, -1, `"result":{"block_id":`},
 		// to keep test simple, allow returning application error in following case
 		{"invalid/missing required param", "/tx", http.StatusOK, int(json2.E_INVALID_REQ), `missing param 'hash'`},
-		{"valid/missing optional param", "/block", http.StatusOK, -1, ``},
+		{"valid/missing optional param", "/block", http.StatusOK, -1, `"result":{"block_id":`},
 		{"valid/included block tag param", "/block?height=included", http.StatusOK, int(json2.E_INTERNAL), "failed to load hash from index"},
 		{"valid/no params", "/abci_info", http.StatusOK, -1, `"last_block_height":"345"`},
 		{"valid/int param", "/block?height=321", http.StatusOK, int(json2.E_INTERNAL), "failed to load hash from index"},
@@ -85,6 +87,19 @@ func TestREST(t *testing.T) {
 	_, local := getRPC(t, "TestREST")
 	handler, err := GetHTTPHandler(local, log.TestingLogger())
 	require.NoError(err)
+
+	// wait for blocks
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+	for {
+		if errors.Is(err, context.DeadlineExceeded) {
+			t.FailNow()
+		}
+		resp, err := local.Header(ctx, nil)
+		if err == nil && resp.Header.Height > 1 {
+			break
+		}
+	}
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {

